tests/resources/ankaios.resource

```diff
@@ -87,125 +85,115 @@ Podman has deleted all existing volumes
-
-the workload "${workload_name}" shall have the execution state "${expected_execution_state}" on agent "${expected_agent_name}"
-    Log    Assert: ${workload_name} shall have execution state ${expected_execution_state}
-    @{list_result}=    table to list    ${CURRENT_RESULT.stdout}
-    &{dict_result}=    table to dict    ${list_result}    WORKLOAD NAME
-    &{item}=    Get From Dictionary    ${dict_result}    ${workload_name}
-    ${actual_execution_state}=    Get From Dictionary    ${item}    EXECUTION STATE
-    ${actual_agent_name}=    Get From Dictionary    ${item}    AGENT
-    Should be equal    ${actual_execution_state}    ${expected_execution_state}
-    Should be equal    ${actual_agent_name}    ${expected_agent_name}
-
 user updates the state "${new_state_yaml_file}" with "${field_value}"
     @{key_val}=    Split String    ${field_value}    separator==
     ${yaml_data}=    Get File    ${new_state_yaml_file}
     ${result_dict}=    Yaml To Dict    ${yaml_data}
-    Log    ${result_dict}
     ${new_config}=    Replace Config    ${result_dict}    filter_path=${key_val}[0]    new_value=${key_val}[1]
-    Log    ${new_config}
     Write Yaml    new_yaml=${new_config}    path=${new_state_yaml_file}
 
-the workload "${workload_name}" has the execution state "${execution_state}"
-    ${result}=    Wait For Execution State
-    ...    %{ANK_BIN_DIR}${/}ank get workloads
-    ...    ${workload_name}
-    ...    ${execution_state}
-    ...    20
-    ...    5
-    Should Be True    ${result}
+the workload "${workload_name}" shall have the execution state "${expected_execution_state}" on agent "${expected_agent_name}" within "${timeout}" seconds
+    @{list_result}=    Create List
+    IF    ${timeout} > 0
+        @{list_result}=    Wait For Execution State
+        ...    %{ANK_BIN_DIR}${/}ank get workloads
+        ...    ${workload_name}
+        ...    ${expected_execution_state}
+        ...    ${timeout}
+    END
+    Should Not Be Empty    ${list_result}    msg=Timeout: Workload '${workload_name}' on agent '${expected_agent_name}' does not have the execution state '${expected_execution_state}'!
 
-the workload "${workload_name}" shall have the execution state "${execution_state}"
-    the workload "${workload_name}" has the execution state "${execution_state}"
+    &{dict_result}=    table to dict    ${list_result}    WORKLOAD NAME
+    &{item}=    Get From Dictionary    ${dict_result}    ${workload_name}
+    ${actual_execution_state}=    Get From Dictionary    ${item}    EXECUTION STATE
+    ${actual_agent_name}=    Get From Dictionary    ${item}    AGENT
+    Should be equal    ${actual_execution_state}    ${expected_execution_state}
+    Should be equal    ${actual_agent_name}    ${expected_agent_name}
 
-the workload "${workload_name}" shall not exist
-    Log    Assert: ${workload_name} shall not exist
+the workload "${workload_name}" shall not exist within "${timeout}" seconds
```

Comment:
It seems like we don't use the timeout here at all. Maybe we can remove it?

Reply:
timeout is handled now